wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/csvCollection/csv/CsvUploadController.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author Samia Benjida <samia.benjida at nist.gov>
+ */
+@RestController
+//@Api(tags="CsvCollection Entity")
+@RequestMapping(CoreConfig.BASE_URI + "/csvCollections/{csvCollectionId}/csv")
+public class CsvUploadController  extends FlowjsController {
+
+    private static final Logger LOG = Logger.getLogger(CsvUploadController.class.getName());
+
+    @Autowired
+    private CsvRepository csvRepository;
+
+    @Autowired
+    private CsvCollectionRepository csvCollectionRepository;
+
+    @Autowired
+    private CoreConfig config;
+
+    @RequestMapping(
+            value = "",
+            method = RequestMethod.GET,
+            params = {
+                    "flowChunkNumber", "flowTotalChunks", "flowChunkSize",
+                    "flowTotalSize", "flowIdentifier", "flowFilename",
+                    "flowRelativePath"})
+    public void isChunckUploaded(
+            @PathVariable("csvCollectionId") String csvCollectionId,
+            HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+        assertCollectionModifiable(csvCollectionId);
+        super.isChunckUploaded(request, response,
+                new CsvCollectionFileParameters(csvCollectionId));
+    }
+
+    @RequestMapping(
+            value = "",
+            method = RequestMethod.POST,
+            params = {
+                    "flowChunkNumber", "flowTotalChunks", "flowChunkSize",
+                    "flowTotalSize", "flowIdentifier", "flowFilename",
+                    "flowRelativePath"})
+    public void uploadChunck(
+            @PathVariable("csvCollectionId") String csvCollectionId,
+            HttpServletRequest request,
+            HttpServletResponse response) throws IOException {
+        assertCollectionModifiable(csvCollectionId);
+        super.uploadChunck(request, response,
+                new CsvCollectionFileParameters(csvCollectionId));
+    }
+
+    private void assertCollectionModifiable(String csvCollectionId) {
+        Optional<CsvCollection> oldTc = csvCollectionRepository.findById(
+                csvCollectionId);
+        if (! oldTc.isPresent()) {
+            throw new ClientException("Collection " + csvCollectionId
+                    + " does not exist.");
+        }
+//        if (oldTc.get().isLocked()) {
```

Comment:
This should probably be uncommented now that locking system is in place for CSV collections